tests/core/test_signal.py

```diff
@@ -250,3 +252,25 @@ async def test_subscription_logs(caplog):
     assert "Making subscription" in caplog.text
     cache.close()
     assert "Closing subscription on source" in caplog.text
+
+
+async def test_signal_lazily_connects(RE):
+    mock_signal_rw = epics_signal_rw(int, "pva://mock_signal", name="mock_signal")
+    await mock_signal_rw.connect(mock=True)
+    RE(ensure_connected(mock_signal_rw, mock=True))
+    assert (
+        mock_signal_rw._connect_task
+        and mock_signal_rw._connect_task.done()
+        and not mock_signal_rw._connect_task.exception()
+    )
+
+
+async def test_signal_mock_and_back_again(RE):
+    mock_signal_rw = epics_signal_rw(int, "pva://mock_signal", name="mock_signal")
```

Comment:
Can you use a `soft_signal_rw` instead please? This will never connect in `mock=False` mode

Reply:
ok sure